service-registry/src/main/java/org/apache/servicecomb/serviceregistry/consumer/MicroserviceManager.java

```diff
@@ -41,11 +47,31 @@ public MicroserviceManager(AppManager appManager, String appId) {
   }
 
   public MicroserviceVersions getOrCreateMicroserviceVersions(String microserviceName) {
-    return versionsByName.computeIfAbsent(microserviceName, name -> {
+    MicroserviceVersions microserviceVersions = versionsByName.computeIfAbsent(microserviceName, name -> {
       MicroserviceVersions instance = new MicroserviceVersions(appManager, appId, microserviceName);
       instance.submitPull();
       return instance;
     });
+    if (!microserviceVersions.isValidated()) {
+      // remove this microservice if it does not exist or not registered in order to get it back when access it again
+      versionsByName.remove(microserviceName);
+    } else {
+      microserviceVersions.updateLastAccessTime();
+    }
+    if (versionsByName.size() >= MAX_NUM_OF_MICROSERVICES) {
```

Comment:
i'm not sure if it is right.
think about edge, if realy need route to so many targets, this logic will cause problem.